run_linux.go

```diff
@@ -258,8 +258,8 @@ rootless=%d
 	}
 
 	defer func() {
-		if err := cleanupRunMounts(mountPoint, runArtifacts); err != nil {
-			options.Logger.Errorf("unabe to cleanup run mounts %v", err)
+		if err = cleanupRunMounts(mountPoint, runArtifacts); err != nil {
+			err = errors.Wrapf(err, "unable to cleanup run mounts")
```

Comment:
Why did you change this?  The err is never going to be reported now?

Reply:
Oh so we just setting already declared `err` from `Run(` its not a new variable. Problem with using options.Logger it becomes `nil` inside `func` and buildah goes into panic state I think it was never caught cause we never got any failures while removing mounts.

Please correct me if this looks wrong I can change this but i think we cannot use `options.Logger.Errorf("unabe to cleanup run mounts %v", err)`

Reply:
we can add a `recover` check or use `fmt.Println(err.Error())` inside the defer instead of `options.Logger.Errorf("unabe to cleanup run mounts %v", err)` if that sounds good.

Reply:
Just for ref: panic logs before this https://cirrus-ci.com/task/6666263280746496?logs=integration_test#L1223

Reply:
@rhatdan reverting this change since it is not relevent to this PR will open different for fixing this in entire code i think we might have multiple instances of this.

Reply:
Remember this message has to get to the remote client on Podman.  Which is why we are using
options.Logger